jupyter-extension/src/plugins/mount/mount.tsx

```diff
@@ -263,6 +287,89 @@ export class MountPlugin implements IMountPlugin {
     this._showDatumSignal.emit(shouldShow);
   };
 
+  saveMountedReposList = (): void => {
+    const mounted = this._poller.mounted;
+    // No mounted repos to save
+    if (mounted.length === 0) {
+      this._repoViewInputSpec = {};
+    }
+    // Single mounted repo to save
+    else if (mounted.length === 1) {
+      const pfsInput: PfsInput = {
+        pfs: {
+          ...(mounted[0].branch !== 'master' && {
+            name: `${mounted[0].repo}_${mounted[0].branch}`,
+          }),
+          repo: mounted[0].repo,
+          ...(mounted[0].branch !== 'master' && {branch: mounted[0].branch}),
+          glob: '/',
+        },
+      };
+      this._repoViewInputSpec = pfsInput;
+    }
+    // Multiple mounted repos to save; use cross
+    else {
```

Comment:
You can loop over mounted by default and check the resulting array length instead of duplicating the pfsInput construction logic.